cmd/kubeadm/app/util/etcd/etcd.go

```diff
@@ -127,6 +124,69 @@ func NewFromCluster(client clientset.Interface, certificatesDir string) (*Client
 	return etcdClient, nil
 }
 
+// retrieveEtcdEndpoints returns the list of etcd endpoints.
+func retrieveEtcdEndpoints(client clientset.Interface) ([]string, error) {
+	etcdEndpoints, err := retrieveEtcdEndpointsFromPodAnnotationWithRetry(client)
+	if err != nil {
+		return retrieveEtcdEndpointsFromClusterStatus(client)
+	}
+	return etcdEndpoints, nil
+}
+
+// retrieveEtcdEndpointsFromPodAnnotation returns the list of endpoints as reported on etcd's pod annotations
+func retrieveEtcdEndpointsFromPodAnnotationWithRetry(client clientset.Interface) ([]string, error) {
```

Comment:
same as above (drop `WithRetry`) + explain why we are waiting in the code; in this case, the main reason is to wait for static pods to get registered, is it right?

Reply:
Yes, in both cases the same reasons apply, protection against transient failures (coming from the apiserver itself, or a load balancer), and to wait for static pod mirroring.